offload/libomptarget/omptarget.cpp

```diff
@@ -1197,6 +1197,97 @@ class PrivateArgumentManagerTy {
   }
 };
 
+/// Try to determine if kernel argument is unused. This method
+/// takes a conservative approach, i.e. it may return false
+/// negatives but it should never return a false positive.
```

Comment:
This deserves some more explanation. Why are these arg types "unused". Basically explain that they have no explicit or implicit variable use in the region.